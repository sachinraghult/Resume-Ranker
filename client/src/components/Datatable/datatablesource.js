import { Link } from "react-router-dom";

export const userColumns = [
  // { field: "id", headerName: "ID", width: 70 },
  {
    field: "user",
    headerName: "User",
    width: 350,
    renderCell: (params) => {
      return (
        <Link
          to={`/preprocessed-profile/${params.row.applicationId}`}
          style={{ textDecoration: "none" }}
        >
          <div className="cellWithImg">
            <img
              className="cellImg"
              src={params.row.img}
              referrerPolicy="no-referrer"
              alt="avatar"
            />
            {params.row.username}
          </div>
        </Link>
      );
    },
  },
  {
    field: "email",
    headerName: "Status",
    width: 150,
    renderCell: (params) => {
      return (
        <div className="cellAction">
          <Link
            to={`/preprocessed-profile/${params.row.applicationId}`}
            style={{ textDecoration: "none" }}
          >
            <div
              className="viewButton"
              style={{
                color:
                  params.row.status === "In Review"
                    ? "orange"
                    : params.row.status === "Accepted"
                    ? "green"
                    : "red",
                border:
                  params.row.status === "In Review"
                    ? "1px dotted orange"
                    : params.row.status === "Accepted"
                    ? "1px dotted green"
                    : "1px dotted red",
              }}
            >
              {params.row.status}
            </div>
          </Link>
        </div>
      );
    },
  },

  // {
  //   field: "status",
  //   headerName: "Status",
  //   width: 150,
  //   renderCell: (params) => {
  //     return (
  //       <div className="cellAction">
  //         <div className="cellWithStatus pending">In Review</div>
  //       </div>
  //     );
  //   },
  // },
  {
    field: "action",
    headerName: "Action",
    width: 150,
    renderCell: (params) => {
      return (
        <div className="cellAction">
          <a
            href={`https://drive.google.com/uc?id=${params.row.resumeId}`}
            target="_blank"
            style={{ textDecoration: "none" }}
          >
            <div className="viewButton">Resume</div>
          </a>
        </div>
      );
    },
  },
];

//temporary data
export const userRows = [
  {
    id: 1,
    username: "Snow",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    status: "active",
    email: "1snow@gmail.com",
    age: 35,
  },
  {
    id: 2,
    username: "Jamie Lannister",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "2snow@gmail.com",
    status: "passive",
    age: 42,
  },
  {
    id: 3,
    username: "Lannister",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "3snow@gmail.com",
    status: "pending",
    age: 45,
  },
  {
    id: 4,
    username: "Stark",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "4snow@gmail.com",
    status: "active",
    age: 16,
  },
  {
    id: 5,
    username: "Targaryen",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "5snow@gmail.com",
    status: "passive",
    age: 22,
  },
  {
    id: 6,
    username: "Melisandre",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "6snow@gmail.com",
    status: "active",
    age: 15,
  },
  {
    id: 7,
    username: "Clifford",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "7snow@gmail.com",
    status: "passive",
    age: 44,
  },
  {
    id: 8,
    username: "Frances",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "8snow@gmail.com",
    status: "active",
    age: 36,
  },
  {
    id: 9,
    username: "Roxie",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "snow@gmail.com",
    status: "pending",
    age: 65,
  },
  {
    id: 10,
    username: "Roxie",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "snow@gmail.com",
    status: "active",
    age: 65,
  },
  {
    id: 1,
    username: "Snow",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    status: "active",
    email: "1snow@gmail.com",
    age: 35,
  },
  {
    id: 2,
    username: "Jamie Lannister",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "2snow@gmail.com",
    status: "passive",
    age: 42,
  },
  {
    id: 3,
    username: "Lannister",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "3snow@gmail.com",
    status: "pending",
    age: 45,
  },
  {
    id: 4,
    username: "Stark",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "4snow@gmail.com",
    status: "active",
    age: 16,
  },
  {
    id: 5,
    username: "Targaryen",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "5snow@gmail.com",
    status: "passive",
    age: 22,
  },
  {
    id: 6,
    username: "Melisandre",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "6snow@gmail.com",
    status: "active",
    age: 15,
  },
  {
    id: 7,
    username: "Clifford",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "7snow@gmail.com",
    status: "passive",
    age: 44,
  },
  {
    id: 8,
    username: "Frances",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "8snow@gmail.com",
    status: "active",
    age: 36,
  },
  {
    id: 9,
    username: "Roxie",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "snow@gmail.com",
    status: "pending",
    age: 65,
  },
  {
    id: 10,
    username: "Roxie",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "snow@gmail.com",
    status: "active",
    age: 65,
  },
  {
    id: 1,
    username: "Snow",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    status: "active",
    email: "1snow@gmail.com",
    age: 35,
  },
  {
    id: 2,
    username: "Jamie Lannister",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "2snow@gmail.com",
    status: "passive",
    age: 42,
  },
  {
    id: 3,
    username: "Lannister",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "3snow@gmail.com",
    status: "pending",
    age: 45,
  },
  {
    id: 4,
    username: "Stark",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "4snow@gmail.com",
    status: "active",
    age: 16,
  },
  {
    id: 5,
    username: "Targaryen",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "5snow@gmail.com",
    status: "passive",
    age: 22,
  },
  {
    id: 6,
    username: "Melisandre",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "6snow@gmail.com",
    status: "active",
    age: 15,
  },
  {
    id: 7,
    username: "Clifford",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "7snow@gmail.com",
    status: "passive",
    age: 44,
  },
  {
    id: 8,
    username: "Frances",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "8snow@gmail.com",
    status: "active",
    age: 36,
  },
  {
    id: 9,
    username: "Roxie",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "snow@gmail.com",
    status: "pending",
    age: 65,
  },
  {
    id: 10,
    username: "Roxie",
    img: "https://images.pexels.com/photos/1820770/pexels-photo-1820770.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500",
    email: "snow@gmail.com",
    status: "active",
    age: 65,
  },
];
