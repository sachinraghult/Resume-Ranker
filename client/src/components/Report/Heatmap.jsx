import React from "react";
import { HeatMapGrid } from "react-grid-heatmap";

const HeatMap = ({ scores }) => {
  const xLabels = ["Skill", "Exp", "Org", "JobSwitch", "Keyword", "Institute"];
  const yLabels = [""];
  const datas = scores
    ? [
        [
          scores[1].toFixed(2),
          scores[2].toFixed(2),
          scores[3].toFixed(2),
          scores[4].toFixed(2),
          scores[5].toFixed(2),
          scores[6].toFixed(2),
        ],
      ]
    : [[0, 0, 0, 0, 0, 0]];
  return (
    <div
      style={{
        width: "100%",
        marginLeft: "80px",
      }}
    >
      <HeatMapGrid
        data={datas}
        xLabels={xLabels}
        yLabels={yLabels}
        // Reder cell with tooltip
        cellRender={(x, y, value) => (
          <div title={`Pos(${x}, ${y}) = ${value}`}>{value}</div>
        )}
        xLabelsStyle={(index) => ({
          //   color: index % 2 ? 'transparent' : '#777',
          fontSize: ".8rem",
        })}
        yLabelsStyle={() => ({
          fontSize: ".8rem",
          textTransform: "uppercase",
          color: "#777",
        })}
        cellStyle={(_x, _y, value) => ({
          background: `rgb(0, 153, 255, ${value + 0.05})`,
          fontSize: ".8rem",
          color: `rgb(0, 0, 0)`,
          marginTop: "5px",
        })}
        // cellStyle={(_x, _y, ratio) => ({
        //   background: `rgb(128, 0, 128, ${ratio})`,
        //   fontSize: ".8rem",
        //   color: `rgb(0, 0, 0, ${ratio / 2 + 0.4})`,
        // })}
        cellHeight="4rem"
        cellWidth="4rem"
        xLabelsPos="bottom"
        yLabelsPos="right"
        square
      />
    </div>
  );
};

export default HeatMap;
