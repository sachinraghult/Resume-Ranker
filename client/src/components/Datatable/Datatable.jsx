import { DataGrid } from "@mui/x-data-grid";
import HeatMap from "./Heatmap";
import "./datatable.css";
import { userColumns, userRows } from "./datatablesource";
import { Link } from "react-router-dom";
import { useLocation } from "react-router-dom";
import { useEffect, useState, useContext } from "react";
import axios from "axios";
import { Context } from "../../context/Context";

import { SERVER_URL } from "../../config";
import { Box, withStyles } from "@material-ui/core";

const StyledDataGrid = withStyles({
  root: {
    "& .MuiDataGrid-renderingZone": {
      maxHeight: "none !important",
    },
    "& .MuiDataGrid-cell": {
      lineHeight: "unset !important",
      maxHeight: "none !important",
      whiteSpace: "normal",
    },
    "& .MuiDataGrid-row": {
      maxHeight: "none !important",
    },
  },
})(DataGrid);

const Datatable = (props) => {
  const { authToken } = useContext(Context);
  const [data, setData] = useState([]);
  const [showData, setShowData] = useState([]);
  const [checked, setChecked] = useState([]);

  const location = useLocation();
  const path = location.pathname.split("/")[2];

  useEffect(() => {
    const fetchCards = async () => {
      try {
        const res = await axios.get(
          `${SERVER_URL}/application/ranking/` + path,
          {
            headers: { Authorization: authToken },
          }
        );

        const overallRanking = [...res.data.result].sort(
          (a, b) =>
            b.preprocessing_data.scores[0] - a.preprocessing_data.scores[0]
        );

        const finalOverallRanking = [...overallRanking, ...res.data.failed];
        setData(finalOverallRanking);
      } catch (err) {}
    };

    fetchCards();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  let datas = [];
  if (props.cat && props.cat != "Select Category" && props.str !== "") {
    var catName;
    if (props.cat === "Skills") catName = "skills";
    else if (props.cat === "Instituition") catName = "college_name";
    else if (props.cat === "Organisation") catName = "company_names";

    const finnd = (data) => {
      if (data.preprocessing_data) {
        let p = data.preprocessing_data[catName];
        return p.find((element) => {
          if (element.toLowerCase() === props.str.toLowerCase()) {
            if (props.searcher && !props.bulk)
              if (!data.isBulk) return true;
              else return false;
            else if (!props.searcher && props.bulk)
              if (data.isBulk) return true;
              else return false;
            else return true;
          }
        });
      }
    };
    data.filter((data) => finnd(data))?.map((data) => datas.push(data));
  } else if (props.searcher && !props.bulk) {
    const finnd = (data) => {
      if (!data.isBulk) return true;
    };

    data.filter((data) => finnd(data))?.map((data) => datas.push(data));
  } else if (!props.searcher && props.bulk) {
    const finnd = (data) => {
      if (data.isBulk) return true;
    };
    data.filter((data) => finnd(data))?.map((data) => datas.push(data));
  } else {
    datas = data;
  }

  const includeList = [];
  if (props.skill) includeList.push(1);
  if (props.exp) includeList.push(2);
  if (props.org) includeList.push(3);
  if (props.jobs) includeList.push(4);
  if (props.keyword) includeList.push(5);
  if (props.ins) includeList.push(6);
  if (includeList.length === 0) includeList.push(0);

  const calculateSum = (a) => {
    if (a.preprocessing_data) {
      let total = 0;
      includeList?.map((i) => (total += a.preprocessing_data.scores[i]));
      return total;
    } else {
      return 0;
    }
  };

  const ranking = [...datas].sort((a, b) => calculateSum(b) - calculateSum(a));

  const handleDelete = (id) => {
    setData(data.filter((item) => item.id !== id));
  };

  const actionColumn = [
    {
      field: "action",
      headerName: "Action",
      width: 200,
      renderCell: (params) => {
        return (
          <div className="cellAction">
            <Link to="/users/test" style={{ textDecoration: "none" }}>
              <div className="viewButton">View Resume</div>
            </Link>
            <div
              className="deleteButton"
              onClick={() => handleDelete(params.row.id)}
            >
              Delete
            </div>
          </div>
        );
      },
    },
  ];
  const rankColumn = [
    {
      field: "rank",
      headerName: "Rank",
      width: 200,
      renderCell: (params) => {
        return <div> {1}</div>;
      },
    },
  ];

  const heatColumn = [];

  const OverallScoreColumn = [
    {
      field: "overall",
      headerName: "Score",
      width: 100,
      renderCell: (params) => {
        return (
          <div>
            <b>
              {params.row.preprocessing_data
                ? params.row.preprocessing_data.scores[0].toFixed(2)
                : 0}
            </b>
          </div>
        );
      },
    },
    {
      field: "heatmap",
      headerName: "Score Distribution",
      width: 350,
      renderCell: (params) => {
        return (
          <div style={{ marginBottom: "10px" }}>
            {params.row.preprocessing_data ? (
              <HeatMap scores={params.row.preprocessing_data.scores} />
            ) : (
              <div style={{ marginTop: "25px", marginBottom: "20px" }}>
                No Data
              </div>
            )}
          </div>
        );
      },
    },
  ];

  return (
    <div className="datatable">
      {/* <div className="datatableTitle">
        Add New User
        <Link to="/users/new" className="link">
          Add New
        </Link>
      </div> */}
      <StyledDataGrid
        height="50px"
        className="datagrid"
        rows={ranking}
        columns={userColumns.concat(OverallScoreColumn)}
        pageSize={10}
        rowsPerPageOptions={[10]}
        // checkboxSelection
      />
    </div>
  );
};

export default Datatable;
