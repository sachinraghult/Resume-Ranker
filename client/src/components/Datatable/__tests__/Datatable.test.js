import React from "react";
global.React = React;
import ReactDOM from "react-dom/client";
import Datatable from "../Datatable";
import Heatmap from "../Heatmap";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";

it("renders without crashing", async () => {
  const root = ReactDOM.createRoot(document.createElement("div"));
  await act(async () =>
    render(
      <BrowserRouter>
        <Datatable />
        <Heatmap />
      </BrowserRouter>
    )
  );
});

it("renders correctly", () => {
  const data1 = [0.37, 0.55, 0.1, 0.83, 0.62, 0.1, 0.1];
  render(
    <BrowserRouter>
      <Datatable />
      <Heatmap scores={data1} />
    </BrowserRouter>
  );
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Datatable />
        <Heatmap />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
