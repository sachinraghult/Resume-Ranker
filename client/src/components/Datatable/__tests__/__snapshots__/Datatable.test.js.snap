// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="datatable"
  >
    <div
      aria-colcount={5}
      aria-multiselectable={false}
      aria-rowcount={0}
      className="datagrid memo-root-3 MuiDataGrid-root MuiDataGrid-root--densityStandard css-r11z79-MuiDataGrid-root"
      role="grid"
    >
      <div
        className="MuiDataGrid-main css-204u17-MuiDataGrid-main"
      >
        <div
          className="MuiDataGrid-overlay css-18rq0oa-MuiDataGrid-overlay"
          error={[TypeError: Cannot set property 'scrollLeft' of null]}
        >
          An error occurred.
        </div>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row-reverse",
          "justifyContent": "flex-end",
        }
      }
    >
      <div
        style={
          Object {
            "marginBottom": "22px",
          }
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "textAlign": "left",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "color": "#777",
                "fontSize": ".8rem",
                "lineHeight": "3rem",
                "padding": "0 0.2rem",
                "textTransform": "uppercase",
              }
            }
          />
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column-reverse",
            "flexGrow": 0,
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "display": "flex",
                "textAlign": "center",
              }
            }
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Skill
            </div>
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Exp
            </div>
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Org
            </div>
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Job
            </div>
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Key
            </div>
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Inst
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 0,
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "initial",
              }
            }
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "background": "rgb(0, 153, 255, NaN)",
                  "border": "1px solid #fff",
                  "borderRadius": "4px",
                  "borderWidth": "1px 1px 0 0",
                  "boxSizing": "border-box",
                  "color": "rgb(0, 0, 0)",
                  "cursor": "initial",
                  "flexBasis": "3rem",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontSize": ".8rem",
                  "height": "3rem",
                  "lineHeight": "3rem",
                  "marginTop": "5px",
                  "overflow": "hidden",
                  "textAlign": "center",
                }
              }
            >
              <div
                title="Pos(0, 0) = 0"
              >
                0
              </div>
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "background": "rgb(0, 153, 255, NaN)",
                  "border": "1px solid #fff",
                  "borderRadius": "4px",
                  "borderWidth": "1px 1px 0 0",
                  "boxSizing": "border-box",
                  "color": "rgb(0, 0, 0)",
                  "cursor": "initial",
                  "flexBasis": "3rem",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontSize": ".8rem",
                  "height": "3rem",
                  "lineHeight": "3rem",
                  "marginTop": "5px",
                  "overflow": "hidden",
                  "textAlign": "center",
                }
              }
            >
              <div
                title="Pos(0, 1) = 0"
              >
                0
              </div>
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "background": "rgb(0, 153, 255, NaN)",
                  "border": "1px solid #fff",
                  "borderRadius": "4px",
                  "borderWidth": "1px 1px 0 0",
                  "boxSizing": "border-box",
                  "color": "rgb(0, 0, 0)",
                  "cursor": "initial",
                  "flexBasis": "3rem",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontSize": ".8rem",
                  "height": "3rem",
                  "lineHeight": "3rem",
                  "marginTop": "5px",
                  "overflow": "hidden",
                  "textAlign": "center",
                }
              }
            >
              <div
                title="Pos(0, 2) = 0"
              >
                0
              </div>
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "background": "rgb(0, 153, 255, NaN)",
                  "border": "1px solid #fff",
                  "borderRadius": "4px",
                  "borderWidth": "1px 1px 0 0",
                  "boxSizing": "border-box",
                  "color": "rgb(0, 0, 0)",
                  "cursor": "initial",
                  "flexBasis": "3rem",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontSize": ".8rem",
                  "height": "3rem",
                  "lineHeight": "3rem",
                  "marginTop": "5px",
                  "overflow": "hidden",
                  "textAlign": "center",
                }
              }
            >
              <div
                title="Pos(0, 3) = 0"
              >
                0
              </div>
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "background": "rgb(0, 153, 255, NaN)",
                  "border": "1px solid #fff",
                  "borderRadius": "4px",
                  "borderWidth": "1px 1px 0 0",
                  "boxSizing": "border-box",
                  "color": "rgb(0, 0, 0)",
                  "cursor": "initial",
                  "flexBasis": "3rem",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontSize": ".8rem",
                  "height": "3rem",
                  "lineHeight": "3rem",
                  "marginTop": "5px",
                  "overflow": "hidden",
                  "textAlign": "center",
                }
              }
            >
              <div
                title="Pos(0, 4) = 0"
              >
                0
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
